circe/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/InterpretExprAsJson.scala

```diff
@@ -0,0 +1,187 @@
+package com.rallyhealth.vapors.factfilter.evaluator
+
+import cats.data.NonEmptyList
+import cats.kernel.Monoid
+import cats.{FlatMap, Foldable, Functor, FunctorFilter, Show, Traverse, TraverseFilter}
+import com.rallyhealth.vapors.core.algebra.Expr
+import com.rallyhealth.vapors.core.data.Window
+import com.rallyhealth.vapors.core.logic.{Conjunction, Disjunction, Negation}
+import com.rallyhealth.vapors.core.math.{Addition, Negative, Subtraction}
+import com.rallyhealth.vapors.factfilter.data.ExtractBoolean
+import com.rallyhealth.vapors.factfilter.data.circe._
+import io.circe.syntax._
+import io.circe.{Json, JsonObject}
+import shapeless.HList
+
+object InterpretExprAsJson {
+
+  type Out[R] = Json
+
+  def encode[F[_], V, R, P](expr: Expr[F, V, R, P]): Json = expr.visit(new InterpretExprAsJson())
+
+  // This takes a higher-kinded parameter because there isn't a good way to use a wild-card
+  // See https://github.com/scala/bug/issues/8039 for more details
+  private[vapors] def commonJsonEncoding[F[_]](
+    expr: Expr[F, _, _, _],
+    additionalFields: Seq[(String, Json)] = Seq.empty,
+  ): Json = {
+    var obj = {
+      JsonObject(
+        "node" -> expr.getClass.getSimpleName.asJson,
+      )
+    }
+    obj = {
+      if (additionalFields.nonEmpty) JsonObject.fromIterable(additionalFields).deepMerge(obj)
+      else obj
+    }
+    Json.fromJsonObject(obj)
+  }
+}
+
+// TODO: Make it optional to stop encoding all subExpressions in the arguments list after a top-level expression?
+//       Or maybe only print the input expression of each node (and arguments will encode all subExpressions)?
+// TODO: Make it optional to unhide Embed node wrappers and just show their sub expression
+// TODO: Make it optional to hide ReturnInput nodes
+class InterpretExprAsJson[F[_], V, P](encodeSubExpressions: Boolean = true)
```

Comment:
There is a lot of TODOs here... I should add unit tests before doing any of these to verify the behavior I want and solidify how things should look.